quinn/src/connection.rs

```diff
@@ -236,9 +220,18 @@ impl FuturesStream for IncomingStreams {
     type Error = ConnectionError;
     fn poll(&mut self) -> Poll<Option<Self::Item>, Self::Error> {
         let mut conn = self.0.lock().unwrap();
-        if conn.closed {
+        if conn.inner.is_drained() {
             return Ok(Async::Ready(None));
         }
+        match conn.pending.error {
```

Comment:
Alternatively, this `poll()` could be a single `if {} else if {} else {}` chain with no explicit `return`s and `if let`s per differently-handled error case, which I think might be slightly more concise and easier to follow?

Reply:
Good call, that actually made it so much easier to follow that I found a minor bug!